File: 30DaysOfCode/day20Sorting.py
import sys


n = int(raw_input().strip())
a = map(int,raw_input().strip().split(' '))

sorted = True
swaps = 0
while (sorted):
    sorted = False
    for j in range(n - 1):
        if (a[j] > a[j + 1]):
            temp = a[j]
            a[j] = a[j + 1]
            a[j + 1] = temp
            swaps += 1
            sorted = True
print "Array is sorted in", swaps, "swaps."
print "First Element:", a[0]
print "Last Element:", a[len(a) - 1]

File: 30DaysOfCode/day20Sorting.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner s = new Scanner(System.in);
        int a = s.nextInt();
        int[] array = new int[a];
        for (int i = 0; i < a; i++) {
            array[i] = s.nextInt();
        }
        int swaps = 0;
        boolean sorted = true;
        while (sorted) {
            sorted = false;
            for (int i = 0; i < a - 1; i++) {
                if (array[i] > array[i + 1]) {
                    int temp = array[i];
                    array[i] = array[i + 1];
                    array[i + 1] = temp; 
                    swaps++;
                    sorted = true;
                }
            }
        }
        System.out.printf("Array is sorted in %d swaps.\n", swaps);
        System.out.printf("First Element: %d\n", array[0]);
        System.out.printf("Last Element: %d\n", array[a - 1]);
    }
}